tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -10,6 +10,58 @@
 #include <linker/sections.h>
 
 
+/*
+ * Offset (starting from the beginning of the vector table)
+ * of the location where the ISRs will be manually installed.
+ */
+#define _ISR_OFFSET 0
+
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+/* The customized solution for nRF52X-based platforms
+ * requires that the POWER_CLOCK_IRQn line equals 0.
+ */
+BUILD_ASSERT_MSG(POWER_CLOCK_IRQn == 0,
+	"POWER_CLOCK_IRQn != 0. Consider rework manual vector table.");
+
+/* The customized solution for nRF52X-based platforms
+ * requires that the RTC1 IRQ line equals 17 and is larger
+ * than the CONFIG_NUM_IRQS.
+ */
+BUILD_ASSERT_MSG(RTC1_IRQn == 17,
+	 "RTC1_IRQn != 17. Consider rework manual vector table.");
+BUILD_ASSERT_MSG(RTC1_IRQn >= CONFIG_NUM_IRQS,
+	 "RTC1_IRQn < NUM_IRQs. Consider rework manual vector table.");
```

Comment:
What is the point in doing this check? `CONFIG_NUM_IRQS` is not used at all when `CONFIG_SOC_SERIES_NRF52X` is defined. Same story for `CONFIG_SOC_SERIES_NRF91X`.

Reply:
Right. This macro reflects the number of ISRs we install, in our case, 3.
I see that it is not used throughout the whole test code, e.g. there are some for loops like:
`i = 0; i < 3; i++
`
i.e. the macro is not used. We can either use "3" in this build-time assert, or just try to use CONFIG_NUM_IRQS in all relevant places in the test code.

Reply:
Fixed that: we are, now using CONFIG_NUM_IRQS throughout the test.